service/task_history.go

```diff
@@ -84,7 +84,16 @@ func (uis *UIServer) taskHistoryPage(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
-	buildVariants := project.GetVariantsWithTask(taskName)
+	repo, err := model.FindRepository(project.Identifier)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	buildVariants, err := task.FindVariantsWithTask(taskName, project.Identifier, repo.RevisionOrderNumber-50, repo.RevisionOrderNumber)
```

Comment:
If someone asks for more than 50 data points on the graph, would it make sense to increase the number of revisions to look back?

Reply:
Possibly, but that can get hairy as you load more revisions. Since this is to find what variants currently have that task, I think a static value would be best